packages/editor-ui/src/stores/workflows.store.ts

```diff
@@ -360,7 +360,7 @@ export const useWorkflowsStore = defineStore(STORES.WORKFLOWS, () => {
 	function getCurrentWorkflow(copyData?: boolean): Workflow {
 		const nodes = getNodes();
 		const connections = allConnections.value;
-		const cacheKey = JSON.stringify({ nodes, connections });
+		const cacheKey = JSON.stringify({ nodes, connections, pinData: pinnedWorkflowData.value });
```

Comment:
Max pin data size I think defaults to 12 MB and can be customized to be even higher. `getCurrentWorkflow` seems to be called from multiple spots. 

Wondering - Could we have performance issues from frequently stringifying a very large JS object? Is there a limit to how long a key can be?

Reply:
There's no limit. I don't have a quick solution here. But removing this caching is much worse for performance.

Reply:
This whole method is performance nightmare, and one of the leading culprits behind UI performance issues. Adding `pinData` isn't going to make it terribly worse than it already is. 

That said, we need to make the `Workflow` class updatable, and update the properties in it, and get rid of this caching code. We can't just keep adding more properties in here. 

At some point we need to own this code across packages, and fix it properly.
I've re-opened the [ticket](https://linear.app/n8n/issue/ENG-123) to fix this.